src/server/services/procedures/daily-word-guess/daily-word-guess.js

```diff
@@ -0,0 +1,157 @@
+ * the corresponding character. Feedback is a "3" if the character is
+ * correct, "2" if it is correct but in the wrong place, and "1" if the
+ * letter is not present in the word.
+ *
+ * @param {BoundedString<5,5>} word Guess for this round
+ */
+DailyWordGuess.guess = async function (word) {
+    word = word.toLowerCase();
+    let realWord = await DailyWordGuess._getDailyWord();
+
+    let prevState = DailyWordGuess._getUserState(this.caller.clientId);
+    
+    // Reject after game over
+    if(prevState.tries <= 0){
+        throw new RPCError('No attempts remaining');
+    }
+
+    WordGuess._validateGuess(word, realWord);
+
+    // Remove a try
+    DailyWordGuess._setUserState(this.caller.clientId, { tries: prevState.tries - 1 });
+
+    return WordGuess._calculateMatches(realWord, word);
+};
+
+/**
+ * Give up on the current game and learn the target word
+ * @returns {String} Target word of daily game
+ */
+DailyWordGuess.giveUp = function () {
+    DailyWordGuess._setUserState(this.caller.clientId, { tries: 0 });
+    return DailyWordGuess._getDailyWord();
+};
+
+/**
+ * The number (out of six) of attempts at the daily puzzle remaining
+ * @returns {Number} Number of attempts remaining
+ */
+DailyWordGuess.triesRemaining = function () {
+    return DailyWordGuess._getUserState(this.caller.clientId).tries;    
+};
+
+/**
+ * The amount of time in hours:minutes:seconds remaining to attempt the current day's puzzle
+ * @returns {String} Amount of time remaining 
+ */
+DailyWordGuess.timeRemaining = function () {
+    const currentTime = new Date();
+    return (23 - currentTime.getHours()).toString().padStart(2,'0') + ':' + (59 - currentTime.getMinutes()).toString().padStart(2,'0') + ':' + (59 - currentTime.getSeconds()).toString().padStart(2,'0');
+};
+
+/**
+ * Get the day's word
+ */
+DailyWordGuess._getDailyWord = async function () {
+    if (DailyWordGuess._dailyWordDate == null || (DailyWordGuess._dailyWordDate.getDate() != new Date().getDate() && DailyWordGuess._dailyWordDate.getMonth() != new Date().getMonth())) {
+        // Check if word of the day already in DB
+        let date = new Date();
+        date.setHours(0, 0, 0, 0);
+
+        let existing = await DailyWordGuess._collection.findOne({
+            date
+        });
+
+        if (existing != null) {
+            // Cache the word of the day
+            logger.log('Existing word found');
+            DailyWordGuess._dailyWord = existing.word;
+            DailyWordGuess._dailyWordDate = new Date(existing.date);
+        } else {
+            logger.log('Generating new word');
+            
+            do {
+                DailyWordGuess._dailyWord = WordGuess._getRandomCommonWord(5);
+
+                let oldDate = new Date(date);
+                oldDate.setFullYear(oldDate.getFullYear() - 1);
+
+                // Test if word already used in past year
+                existing = await DailyWordGuess._collection.findOne({
```

Comment:
I think this actually might not behave as desired when we introduce concurrency. It might be nice to write a test for it. My concern is that if you call two of these (in "parallel") that they might insert two different daily words in the database and the word selected by the first call will be overwritten by the second. Something like:
```javascript
const currentWordList = await Promise.all([
  service._getDailyWord(),
  service._getDailyWord()
]);
assert.equal(...currentWordList);  // I suspect this will fail
```

Reply:
This was a problem that needed to be addressed back when I created the cloud variables service. The easy way to work around it is to switch from `insertOne` to `findOneAndUpdate` using `$setOnInsert` and `upsert`.